crates/api_models/src/subscription.rs

```diff
@@ -149,17 +163,21 @@ pub struct PaymentResponseData {
     pub amount: MinorUnit,
     pub currency: api_enums::Currency,
     pub connector: Option<String>,
+    pub payment_method_id: Option<Secret<String>>,
+    pub payment_experience: Option<api_enums::PaymentExperience>,
+    pub error_code: Option<String>,
+    pub error_message: Option<String>,
 }
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
 pub struct ConfirmSubscriptionRequest {
     /// Client secret for SDK based interaction.
     pub client_secret: Option<String>,
 
     /// Amount to be charged for the invoice.
-    pub amount: MinorUnit,
+    pub amount: Option<MinorUnit>,
```

Comment:
If this is create separate and confirm separate scenario, the confirm call which is a client call which should not control the amount and currency, so lets remove those fields